indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocateAction.java

```diff
@@ -272,8 +273,12 @@ private SegmentIdWithShardSpec tryAllocateFirstSegment(TaskActionToolbox toolbox
   {
     // No existing segments for this row, but there might still be nearby ones that conflict with our preferred
     // segment granularity. Try that first, and then progressively smaller ones if it fails.
+    // Consider trying with WEEK granularity only if the preferred granularity is WEEK
     final List<Interval> tryIntervals = Granularity.granularitiesFinerThan(preferredSegmentGranularity)
```

Comment:
+1, add a new method named `Granularity.nonWeekgranularitiesFinerThan()` or something.

Reply:
Yep, I was hesitant about changing the existing method because it seemed like it would make the utility unintuitive.
Although it's not being used anywhere else